src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DataProviderPass.php

```diff
@@ -52,6 +53,11 @@ private function registerDataProviders(ContainerBuilder $container, string $type
         $queue = new \SplPriorityQueue();
 
         foreach ($services as $serviceId => $tags) {
+            $definition = $container->getDefinition($serviceId);
+            if (is_subclass_of($definition->getClass(), SerializerAwareDataProviderInterface::class)) {
```

Comment:
Btw why not `is_a($definition->getClass(), SerializerAwareDataProviderInterface::class, true)`?